Basic editor functionality

bahl24 opened the pull request:
### Summary of Changes
1. Added basic setup for vue based drag and drop project.
2. 'Add' & 'Remove' button
3. Added v-bind prop in 'class' for BS4 based size.


### Testing Instructions
1. cd administrator/components/com_templates/
2. npm install
3. npm run serve

![Peek 2019-05-20 18-16](https://user-images.githubusercontent.com/36095178/58023163-c1b2aa80-7b2c-11e9-9437-422a1d74b551.gif)

Commit:
[4.0] Code cleanup: remove unnecessary onContentAfterSave in PlgFieldsRepeatable (#24506)

brianteeman committed:
[4.0] New User form switcher (#24942)

Update the code for displaying a swicther on the admin new user form /index.php?option=com_users&view=user&layout=edit

brianteeman committed:
[4.0] switcher legend alignment (#24802)

Make sure that the legend is left aligned in all cases

ReLater committed:
[4.0] Backend. Tabs empty in com_config if module User Status deactivated. (#24928)

* Add HTMLHelper::_('bootstrap.framework')

* Move defined('_JEXEC') up

SharkyKZ committed:
Make login module IDs unique (#24933)

roland-d committed:
[4.0.] Remove obsolete and outdated trigger (#24941)

* Remove obsolete and outdated trigger

Signed-off-by: Roland Dalmulder <contact@rolandd.com>

* Use the correct trigger

Signed-off-by: Roland Dalmulder <contact@rolandd.com>

Arpit-24 committed:
[4.0] Fix to modal not opening on edit button click (#24349)

* Fix overlap of editor and options

Hides the header and options in full screen mode to prevent overlap over the editor area in full-screen mode.

Another possible alternative is displaying both the options as well as the editor in full-screen mode but purely a user preference. Because the need for save and other options is not as important in full-screen mode.

* Move tinymce change to it own file

* Move tinymce change to it own file

* Restore original template.scss

* Restore original file

* Replace invalid &quot; character with " in JS

Fixes issues with running invalid javascript code due to replacement of " by &quot;

* Remove new line at the end

* Remove extra tabs

* Restore File

Commit:
Change the lookup order for the favicon.ico (#17677)

amitranjan2 committed:
fix (#23724)

zero-24 committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

Hackwar committed:
Adding Postgres to system tests (#24949)

brianteeman committed:
[4.0] Privacy filter widths (#24954)

This PR allows the filters to get the correct width

zero-24 committed:
reset drone yml

anuragteapot commented:
@bahl24 See how media manager is integrated with Joomla 4.  https://github.com/joomla-projects/gsoc19_page-builder/tree/4.0-dev/administrator/components/com_media

anuragteapot commented:
i am wondering how we are going to manage different column using this library https://sortablejs.github.io/Vue.Draggable/#/simple

Can we use https://jbaysolutions.github.io/vue-grid-layout/examples/01-basic.html ? 

@bahl24

bahl24 commented:
> @bahl24 See how media manager is integrated with Joomla 4. https://github.com/joomla-projects/gsoc19_page-builder/tree/4.0-dev/administrator/components/com_media

Yes, I am yet to work in the integration part

brianteeman committed:
[4.0] Correct code comments (#24974)

Spotted this copy/paste error when trying to debug #24973

Can be merged on review I hope

chmst committed:
Use textfield for descriptions in workflow views (#24958)

brianteeman committed:
[4.0] Remove accordion (#24910)

* [4.0] Remove accordion

The Upload and Update tab of com_joomlaupdate displays any warnings in an unstyled bootstrap accordion. This PR removes the accordion and displays the warnings in the same way as com_installer&view=warnings it also removes the fieldset wrapper on the warnings as it is not a fieldset and therefore changes the legend to a heading

* Update administrator/components/com_joomlaupdate/tmpl/joomlaupdate/default_upload.php

Co-Authored-By: Quy <quy@fluxbb.org>

* Update administrator/components/com_joomlaupdate/tmpl/joomlaupdate/default_upload.php

Co-Authored-By: Quy <quy@fluxbb.org>

brianteeman committed:
[4.0] Update and improve installer warningss (#24909)

The installer warnings are displayed as a pair of a title and description in com_installer&view=warnings. Because they are a pair some of the text is being repeated. This PR fixes that. See the example below

brianteeman committed:
[4.0] com_banners sidebar (#24869)

* [4.0] com_banners sidebar

removes unused sidebar code from com_banners

* Update administrator/components/com_banners/tmpl/clients/default.php

Co-Authored-By: Quy <quy@fluxbb.org>

* Update administrator/components/com_banners/tmpl/tracks/default.php

Co-Authored-By: Quy <quy@fluxbb.org>

* Update administrator/components/com_banners/tmpl/banners/default.php

Co-Authored-By: Quy <quy@fluxbb.org>

* thanks @quy

chmst committed:
Fix ordering of workflows via column header "state" (#24864)

twister65 committed:
[4.0] database importer exporter (rebase) (#24733)

* Database exporter importer

Rebase PR #23320

* Classname

* Add Database Console Application

* Register Export and Import Commands

* Add commands paths

* Use Joomla Console

php cli/joomla.php

* Database Joomla/Console Application

To import in a new database

* Remove DatabaseConsole

* Sort in alpha order.

zero-24 committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

wilsonge committed:
[4.0] Fix showon error in com_menus (#24996)

* Fix showon error in com_menus

* Fix codestyle

brianteeman committed:
[4.0] Switch admin language button (#24983)

Simple PR that disables the button until a language is selected

laoneo committed:
Remove the deprecate message in the editor class (#24987)

laoneo committed:
Remove the deprecated profiler variable (#24986)

infograf768 committed:
[4.0] Correcting switcher alignment when col is md-3 (#24993)

wilsonge committed:
Remove jquery from modals field (#25002)

brianteeman committed:
[4.0] replace btn-xs with btn-sm (#24979)

In Bootstrap 4 they "dropped the .btn-xs class entirely as .btn-sm is proportionally much smaller than v3’s."

A previous PR added the css back for the frontend template.

This PR removes that additional class and instead replaces the uses of btn-xs with btn-sm

Fixes #21244

wilsonge committed:
[4.0] Fix saving an item in a modal (#25001)

brianteeman committed:
[4.0] Option tabs descriptions (#24984)

Somewhere along the way we lost the display of the fieldset descriptions in the component options. This puts it back. There may be other accessibility issues with this view but it is beyond the scope of this PR

alikon committed:
[4.0] - Postgresql save article (#24148)

zero-24 committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

Hackwar committed:
Updating Codeception to 3.0 (#24970)

zero-24 committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev
